sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileAsyncClient.java

```diff
@@ -1183,7 +1183,6 @@ Mono<Response<Void>> appendWithResponse(Flux<ByteBuffer> data, long fileOffset,
     /**
      * Flushes (writes) data previously appended to the file through a call to append.
      * The previously uploaded data must be contiguous.
-     * <p>By default this method will not overwrite existing data.</p>
```

Comment:
I'm a bit hesitant on making this change as this changes default behavior for an existing API, and the behavioral change will implicitly work without any service errors or warnings. Or, better put, this change will implicitly now have Datalake files being overwritten by default when `flush` is called and this will overwrite data and won't result in any service errors (aside credential or transient errors).

I understand the reasoning behind this is that between releases this began resulting in 412 exceptions as there was a fix to the API that this called to properly include an HTTP precondition when `overwrite` was false (which is the correct fix).

Reply:
Update to my comment, I was told that Datalake files append and flush work differently than block blob put block and commit block list, which is what I was basing my understanding on. Datalake files have to be created before being appended which results in them having an ETag associated with them, whereas block blobs don't require the same so when commit block list is called with `If-None-Match: *` (behavior when overwrite is false, also the behavior of `flush(long)`) it works whereas with a Datalake file the file has an ETag and this will always fail.

Given that the plan forward is to deprecate this API as it doesn't functionally make sense as it'll never work correctly.